### PR TITLE
Switch from Node Test Runner to Jest for compatibility

### PR DESCRIPTION
**Pull request by lemilonkh**
- **Install jest**
- **Setup and configure jest**
- **Modify model tests to work with Jest**
- **Remove unused config segments from jest.config.ts**
- **Fix ESM module issues with jest by adjusting npm run test script**
- **Fix ESM module issue in helpers.ts with import.meta.url**
- **Convert city test to Jest**

### CODE REVIEW

**Comment by evanp** [comment]
-1

What's going on here? Why would we make this kind of sweeping change?

**Comment by thehighestprimenumber** [comment]
I have moved as many tests as I could to jest. Files ending in .jest.ts and are picked up by jest. Files  ending in .test.ts are picked up by the node test runner.

**Review by thehighestprimenumber — APPROVED** [review]

File: app/src/util/api.ts [hunk 1]
```diff
@@ -12,8 +12,6 @@ import { db } from "@/models";
 import { ValidationError } from "sequelize";
 import { ManualInputValidationError } from "@/lib/custom-errors/manual-input-error";
 
-import "@/util/big_int_json.ts";
```
[the remark this answers is not in the document]
Comment (thehighestprimenumber) [reply app/src/util/api.ts]: it's a duplicate import, we already have it on line 1

File: app/tests/api/datasource.jest.ts [hunk 2]
```diff
@@ -55,24 +53,22 @@ describe("DataSource API", () => {
...
-      where: { year: inventoryData.year },
-    });
-    await db.models.DataSource.destroy({
-      where: {
-        [Op.or]: [literal(`dataset_name ->> 'en' LIKE 'XX_INVENTORY_TEST_%'`)],
-      },
+
+    await db.models.Inventory.destroy({ where: { year: inventoryData.year } });
+    await cascadeDeleteDataSource({
+      [Op.or]: [literal(`dataset_name ->> 'en' LIKE 'XX_INVENTORY_TEST_%'`)],
```
[the remark this answers is not in the document]
Comment (thehighestprimenumber) [reply app/tests/api/datasource.jest.ts:63]: Because  support for literal replacements in the `where` object has been removed.

**Pull request merged**